OpenSim/Tools/MarkerPlacer.cpp

```diff
@@ -242,6 +243,47 @@ bool MarkerPlacer::processModel(Model* aModel,
     /* Load the static pose marker file, and average all the
     * frames in the user-specified time range.
     */
+    TimeSeriesTableVec3 staticPoseTable{aPathToSubject + _markerFileName};
+    const auto& timeCol = staticPoseTable.getIndependentColumn();
+    auto numRowsInRange = std::count_if(timeCol.cbegin(),
+                                        timeCol.cend(),
+                                        [&] (const double& val) {
+                                            return val >= _timeRange[0] &&
+                                                   val <= _timeRange[1];
+                                        });
+    const auto avgRow = staticPoseTable.averageRow(_timeRange[0],
+                                                   _timeRange[1]);
+    for(int r = staticPoseTable.getNumRows() - 1; r >= 0; --r) {
+        if(staticPoseTable.getIndependentColumn()[r] >= _timeRange[0] &&
+           staticPoseTable.getIndependentColumn()[r] <= _timeRange[1]) {
+            if(numRowsInRange > 1) {
+                staticPoseTable.removeRowAtIndex(r);
+                --numRowsInRange;
+            } else {
+                staticPoseTable.updRowAtIndex(r) = avgRow;
+            }
+        }
+    }
+    OPENSIM_THROW_IF(!staticPoseTable.hasTableMetaDataKey("Units"),
+                     Exception,
+                     "MarkerPlacer::processModel -- Marker file does not have "
+                     "'Units'.");
+    Units
+    staticPoseUnits{staticPoseTable.getTableMetaData<std::string>("Units")};
+    double scaleFactor = staticPoseUnits.convertTo(aModel->getLengthUnits());
+    OPENSIM_THROW_IF(SimTK::isNaN(scaleFactor),
+                     Exception,
+                     "Model has unspecified units.");
+    if(std::fabs(scaleFactor - 1) >= SimTK::Eps) {
+        for(unsigned r = 0; r < staticPoseTable.getNumRows(); ++r)
+            staticPoseTable.updRowAtIndex(r) *= scaleFactor;
+
+        staticPoseUnits = aModel->getLengthUnits();
+        staticPoseTable.removeTableMetaDataKey("Units");
+        staticPoseTable.addTableMetaData("Units",
+                                         staticPoseUnits.getAbbreviation());
+    }
+    
     MarkerData* staticPose = new MarkerData(aPathToSubject + _markerFileName);
     staticPose->averageFrames(_maxMarkerMovement, _timeRange[0], _timeRange[1]);
     staticPose->convertToUnits(aModel->getLengthUnits());
```

Comment:
Apologies for the confusing comment.  The main point is that although we still create the `staticPose` `MarkerData*` it is not used for anything, so modifying its units and averaging its rows has no effect as far as I can see. It appears that `staticPose` is used only for its markerNames (L294) but that could be replaced by `staticPoseTable.getColumnLabels()`. 

To fully replicate the previous behavior we should be updating `staticPoseTable` (or creating another table) such that it has a single (average) row. I believed that current `scaleTest` (that employs the `MarkerPlacer` still passed because the first row of the `staticPoseTable` was sufficiently close to the average row so that the answers were within the tolerance. But checking it now, there is no regression test for marker positions from `MarkerPlacer` so the answer could be significantly different- there is no way of knowing atm. I have no issue with the `averageRow` implementation, but with the averaging step is now missing altogether. 

We should verify that static pose markers given to the `MarkersReference` (for the IK Solver) remains the same or otherwise verify the results have not changed.  This could be done by comparing the markers on the processed model (after scaling and marker placer) to a standard model. That is the markers locations match the previous (standard) locations. I am happy to help with that too.